sources/osgShadow/ShadowTexture.js

```diff
@@ -77,8 +77,9 @@ define( [
 
             }.bind( this ) );
 
-
+            // tells shader where to find it
```

Comment:
Note: that's actually the piece of code that cause problem. 
(as the compiler will make a uniform node out if it which will conflict with the sample2D uniform)